### PR TITLE
Add a core.online provider to monitor online status

### PR DESCRIPTION
**Pull request by bemasc**
This is essentially a workaround for https://crbug.com/114475

### CODE REVIEW

File: interface/core.online.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "core.battery",
```
Comment (soycode) [inline interface/core.online.json]: Should be core.online.

Comment (bemasc) [inline interface/core.online.json]: Fixed.

**Comment by bemasc** [comment]
I added a comment in answer to your question.  PTAL.

**Comment by soycode** [comment]
LGTM :+1: - and to verify I understand your comment, it sounds like this will still be useful in some cases even after the bug is resolved. I also assume we want to expose this in freedom-for-* flavors and cut new releases, let me know if you'd like me to do that or if you'll send pull requests. Thanks!

**Comment by bemasc** [comment]
> it sounds like this will still be useful in some cases even after the bug is resolved

Yeah ... maybe.  There's no event to listen for, so I guess you'd have to poll `navigator.onLine`, unless there's some tricky solution using `Object.observe()` or something.  Also, the bug has been open for four years and no one's working on it.

**Comment by bemasc** [comment]
> I also assume we want to expose this in freedom-for-* flavors and cut new releases, let me know if you'd like me to do that or if you'll send pull requests.

It would definitely be helpful if you could handle the release sequence.  Thanks!

**Pull request merged**